### PR TITLE
feat: print bundled Node and Electron versions in `cypress version`

### LINKED ISSUE #9180: Report bundled Node version
When building `cypress/included:x.y.z` Docker image sometimes we miss the Node version and create a Docker image with Node version that does not match the Node version bundled with Cypress. It would be nice to validate the Node version during the CI build process. The problem is we do not output bundled Node version from Cypress.

Proposal:

Extend the `cypress version` command to output bundled Node version

```
$ npx cypress version
Cypress package version: 5.4.0
Cypress binary version: 5.4.0
Cypress includes Node: 12.14.1
```

Stretch goal:

For simple parsing add a parameter to output just a particular version

```
$ npx cypress version --package
5.4.0
$ npx cypress version --node
12.14.1
```

### PR DESCRIPTION
**Pull request by bahmutov**
- Closes #9180 

### User facing changelog

When reporting version, Cypress will report Electron and bundled Node versions, something like

```
$ npx cypress version
Cypress package version: 6.0.0
Cypress binary version: 6.0.0
Electron version: 10.1.2
Bundled Node version: 12.16.3
```

### Additional details

This makes building Docker images with matching Node version easier. Probably makes debugging issues easier too, since we know the Electron and Node versions.

### How has the user experience changed?

before
```
$ npx cypress version
Cypress package version: 6.0.0
Cypress binary version: 6.0.0
```

after
```
$ npx cypress version
Cypress package version: 6.0.0
Cypress binary version: 6.0.0
Electron version: 10.1.2
Bundled Node version: 12.16.3
```

### PR Tasks

- [x] Have tests been added/updated?
- [x] Has the original issue or this PR been tagged with a release in ZenHub? 
- [x] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? https://github.com/cypress-io/cypress-documentation/pull/3324

### CODE REVIEW

**Comment by bahmutov** [comment]
Confirming the built Mac binary includes Electron and Electron Node versions 

First install
```
$ CYPRESS_INSTALL_BINARY=https://cdn.cypress.io/beta/binary/6.0.0/darwin-x64/circle-include-electron-node-version-bc2e168861a9882f6a256cd392d864f84c9acfb7/cypress.zip npm i https://cdn.cypress.io/beta/npm/6.0.0/circle-include-electron-node-version-bc2e168861a9882f6a256cd392d864f84c9acfb7/cypress.tgz
```

Second, find the cache

```
$ npx cypress cache path
/Users/gleb/Library/Caches/Cypress
```

Third, print the `package.json` from the binary

```
$ cat /Users/gleb/Library/Caches/Cypress/6.0.0/Cypress.app/Contents/Resources/app/package.json 
{
  "name": "cypress",
  "productName": "Cypress",
  "description": "Cypress.io end to end testing tool",
  "version": "6.0.0",
  "electronVersion": "10.1.5",
  "electronNodeVersion": "12.16.3",
  "main": "index.js",
  "env": "production"
}
```

File: yarn.lock [hunk 1]
```diff
@@ -16525,6 +16540,12 @@ has-ansi@^2.0.0:
   dependencies:
     ansi-regex "^2.0.0"
 
+has-binary2@cypress-io/has-binary#8580a33df21e8b36a43f57872a82c60829636a92:
```
Comment (flotwig) [inline yarn.lock:16543]: gleb, plz delete your yarn cache so you don't have to suffer, it's hurting me to watch this

Comment (bahmutov) [reply yarn.lock:16543]: #%&*##%$#$%%$ this yarn cache lock thing with has-binary2

**Comment by flotwig** [comment]
can you show me where in CI this is printed for Linux/Mac/Windows?

File: yarn.lock (hunk 1 above)
Comment (flotwig) [reply yarn.lock:16543]: for me, `rm -rf ~/.cache/yarn/` was all i needed to do. maybe it's different on mac

**Comment by bahmutov** [comment]
Example on Linux: https://app.circleci.com/pipelines/github/cypress-io/cypress/14662/workflows/4858db03-2372-40db-8135-a38d9692febf/jobs/509850

File: yarn.lock (hunk 1 above)
Comment (bahmutov) [reply yarn.lock:16543]: by deleting the yarn cache you mean the global yarn cache on my machine?

Comment (flotwig) [reply yarn.lock:16543]: yeah, that's where it's located for me

Comment (bahmutov) [reply yarn.lock:16543]: and then "normal" yarn install in the monorepo would not bring the craziness back?

**Comment by bahmutov** [comment]
I have also added printing the version commands to Linux/Mac/Windows to https://github.com/cypress-io/cypress-test-example-repos repo commands

**Comment by bahmutov** [comment]
added printing Cypress version to building Docker `cypress/included` images

File: yarn.lock (hunk 1 above)
Comment (bahmutov) [reply yarn.lock:16543]: Let's try deleting the cache
```
$ yarn cache dir
/Users/gleb/Library/Caches/Yarn/v6
~/git/cypress-docker-images on master
$ rm -rf /Users/gleb/Library/Caches/Yarn/v6
```

Comment (bahmutov) [reply yarn.lock:16543]: ok, removed the cache - do I need to rebuild `yarn.lock` file here for this PR?

Comment (flotwig) [reply yarn.lock:16543]: yeah, this has the bad version of has-binary

**Comment by bahmutov** [comment]
Screen Shot 2020-11-13 at 1 34 34 PM

whyyyyyyyy 😡

**Review by flotwig — APPROVED** [review]
nice

**Pull request merged**